File: BuilderDesignPattern/app/src/main/java/com/btk/builderdesignpattern/MainActivity.java
package com.btk.builderdesignpattern;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        User user = new User.builder()
                .userId("100")
                .name("username")
                .address("User Address")
                .age(30)
                .build();

        Log.v("===","Name:"+user.getName());
        Log.v("===","Address:"+user.getAddress());
        Log.v("===","Age:"+user.getAge());
        Log.v("===","UserId:"+user.getUserId());
    }
}

File: app/src/main/java/com/mnj/builderdesignpattern/MainActivity.kt
package com.mnj.builderdesignpattern

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val user = User.UserBuilder("abc", "xyz").age(10).address("address").build()

        println("==>> User: $user")

        val usr1 = User.UserBuilder("qwe", "rty").build()
        println("==>> User1: $usr1")

        val user2 =
            User.UserBuilder("ghj", "tii").age(25).phone("387843").address("This is address")
                .build()
        println("==>> user2 : $user2")

    }
}